src/index.ts

```diff
@@ -3,14 +3,30 @@ import {
     TransactContext,
     TransactHookResponseType,
     TransactHookTypes,
+    Transaction,
 } from '@wharfkit/session'
 
 /** Import JSON localization strings */
 import defaultTranslations from './translations.json'
 
-export class TransactPluginTemplate extends AbstractTransactPlugin {
+const START_CHECKING_FINALITY_AFTER = 150000 // 2.5 minutes
+
+interface TransactPluginFinalityCallbackOptions {
+    onFinalityCallback: () => void
+}
+
+export class TransactPluginFinalityCallback extends AbstractTransactPlugin {
+    onFinalityCallback: () => void
+
+    constructor({onFinalityCallback}: TransactPluginFinalityCallbackOptions) {
```

Comment:
Probably overkill to use an object here, since there's only one thing you can pass in. 

Do you imagine there will be other options in the future? If so this is fine, otherwise maybe we just pass the callback in?

Reply:
Yeah, I'll make the  START_CHECKING_FINALITY_AFTER value configurable as you suggested. Down the line, we could probably add other options here (like for example polling interval and frequency).